### PR TITLE
Add ranger contribution layer

### PR DESCRIPTION
**Pull request by ralesi**
Adding a newly created package I created, knowing how some spacemacs users (like @trishume) feel about ranger.

### CODE REVIEW

**Comment by felipe-barros** [comment]
Ranger is necessary to life, my most beloved tool. I'll check this out tomorrow. Coincidently I was playing with evil and dired ranger implementations today and writing notes so I can create my own one day.

**Comment by felipe-barros** [comment]
So thank you for your effort :)

**Comment by trishume** [comment]
ZOMG :heart_eyes_cat: 

I've been taking a Spacemacs vacation recently and trying out Atom but this is tempting me to quit early and come back to Spacemacs.

**Comment by cpaulik** [comment]
I've been waiting for that ever since I saw it on reddit. Great work!

**Comment by geo7** [comment]
I remember using ranger once and never bothered opening it again... these reactions make me feel as though I should have given it more time.

**Comment by trishume** [comment]
@geo7 you probably had a decent reason, plain ranger is not *that* impressive, but it's pretty nice. As a student with a hobby of dotfile tweaking I had the time to add all the features I wanted though, so my Ranger is a bit better than most peoples:

https://github.com/trishume/macranger

**Comment by geo7** [comment]
@trishume ah fair... yeah I'd seen people mention it before but never found any benefit over just ls, tree, find or what not.

**Comment by trishume** [comment]
@geo7 if your benchmark is unix tools then definitely use ranger. It's so much faster than ls/cd for navigating and messing with things. You can also just run it from your terminal and set it up so that when you hit `q` to close it your terminal will be `cd`'d to the directory you browsed to.

**Comment by geo7** [comment]
@trishume I'll have to give it another shot at some point then :)

**Comment by ralesi** [comment]
Any chance you can post you dotfile so I have some ideas?

**Comment by catesandrew** [comment]
+1

**Comment by ralesi** [comment]
What tweaks,  anything I can change in the package?

**Comment by robbyoconnor** [comment]
@ralesi -- :heart: :joy_cat:

**Comment by fbob** [comment]
@ralesi OMG ... Awesome ...Terrific ... this package is a highway for Spacemacs world domination ... 
Thanks !!!

**Comment by ralesi** [comment]
Thanks for the comments,  if anyone has any bugs or feature requests,  let me know.

File: contrib/!tools/ranger/Readme.org [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+#+BEGIN_SRC elisp
+(setq evil-ranger-max-preview-size 10)
+#+END_SRC
+
+* Key Bindings
+
+| Key Binding | Description                                 |
+|-------------+---------------------------------------------|
+| ~SPC a r~   |                                             |
```
Comment (person808) [inline contrib/!tools/ranger/Readme.org]: Missing description

**Comment by ralesi** [comment]
@person808 .  Thanks, I updated the readme.

**Comment by robbyoconnor** [comment]
<3 <3

**Comment by ralesi** [comment]
Just added history logging and search through history with `H`

**Comment by robbyoconnor** [comment]
1 file rename: Readme.org -> README.org

It's inconsistent with everything else :)

**Comment by ralesi** [comment]
@robbyoconnor, done and done

**Comment by syl20bnr** [comment]
Pretty neat ! :heart: 
Do you think you can make it behave more like ranger by putting the focus on the middle window instead of the right window ?
I tried to enable the preview via the variable but it did not work, what do I need to do to get the previews on the currently selected file ?

Thank you again :-)
Cherry-picked into `develop` branch, you can safely delete your branch.

**Comment by ralesi** [comment]
@syl20bnr,  can you elaborate?  The primary dired window (rightmost dired window)  should be the only one to have focus.  Press "i"  to show a preview of selected file.

**Comment by syl20bnr** [comment]
@ralesi In ranger the focused window is the centered one, the right most window is the preview (content of the currently selected directory in the centered window, or the content of the file).

capture d ecran 2015-07-27 a 00 05 19

In the screenshot the focus is in the centered column and the right panel is the preview.

**Comment by syl20bnr** [comment]
@ralesi nevermind the following settings made it:

``` elisp
    (setq ranger-parent-depth 1
          ranger-preview-file t)
```

This package is amazing :+1: 

I will push this config to Spacemacs :-)

**Pull request closed without merging**